tests/simple/simple_create.go

```diff
@@ -3,36 +3,147 @@ package simple
+							key, col, err[0]))
+					return nil, maskAny(err[0])
+				} else {
+					t.log.Errorf("Failed to read(%d) document %s in %s (&v).", i, key, col, err)
+					return nil, nil
+				}
+			} else { // document found
+				return result, nil
+			}
+		}
+
+		time.Sleep(backoff)
+		backoff += backoff
+	}
+
+	t.log.Errorf("Timed out while trying to read(%d) document %s in %s (&v).", i, key, col)
+	return nil, maskAny(fmt.Errorf("Timed out while trying to read(%d) document %s in %s (&v).", i, key, col))
+
+}
+
+func (t *simpleTest) createDocument(c *collection, document UserDocument, key string) (string, error) {
+
+	operationTimeout := t.OperationTimeout
+	testTimeout := time.Now().Add(operationTimeout * 4)
+
 	q := url.Values{}
 	q.Set("waitForSync", "true")
-	t.log.Infof("Creating document '%s' in '%s'...", key, c.name)
-	resp, err := t.client.Post(fmt.Sprintf("/_api/document/%s", c.name), q, nil, document, "", nil, []int{200, 201, 202, 409}, []int{400, 404, 307}, operationTimeout, retryTimeout)
-	if err != nil {
-		// This is a failure
-		t.createCounter.failed++
-		t.reportFailure(test.NewFailure("Failed to create document with key '%s' in collection '%s': %v", key, c.name, err))
-		return "", maskAny(err)
-	} else if resp.StatusCode == 409 {
-		// Duplicate key, check if this is correct
-		if rev, err := t.readExistingDocument(c, key, "", true, true); err != nil {
-			// Document with reported duplicate key cannot be read, so 409 status is a failure
+	url := fmt.Sprintf("/_api/document/%s", c.name)
+	backoff := time.Millisecond * 250
+	i := 0
+
+	for {
+
+		i++
+		if time.Now().After(testTimeout) {
+			break;
+		}
+
+		checkRetry := false
+		success := false
+
+		t.log.Infof("Creating (%d) document '%s' in '%s'...", i, key, c.name)
+		resp, err := t.client.Post(url, q, nil, document, "", nil,
+			[]int{0, 1, 200, 201, 202, 409, 503},	[]int{400, 404, 307}, operationTimeout, 1)
+
+/*
+	POST /_api/document
+	{"body":1}
+	-->
+	20x, if document was inserted
+	400, if body bad
+	404, if collection does not exist
+	409, if document already existed
+	timeout, in which case the document might or might not have been written
+		after this, either one of the 4 things might have happened,
+		or nothing might have happened at all,
+		or any of this might still happen in the future
+	connection refused with coordinator ==> simply try again with another
+	connection error ("broken pipe") with coordinator, to be treated like
+		a timeout
+	503, cluster internal mishap, all bets off
+	If first request gives correct result: OK
+	if wrong result: ERROR  (include 503 in this case)
+	if connection refused to coordinator: simply retry other
+	if either timeout (or broken pipe with coordinator):
+		try to read the document repeatedly for up to 15s:
+			if document there: treat as if insert had worked
+		else (not appeared within 15s):
+			treat as if insert has not worked and go to retry loop
+*/
+// createDocument creates a new document.
```

Comment:
This comment must go up again before the function, otherwise the godoc is unhappy.